### PR TITLE
M0078DDG-467 Add Kernel in Module Natures page

### PR DESCRIPTION
**Pull request by tdelhomenie-microej**
- I kept the alphabetical order for the Module Natures
- I did not factorized the properties with the Standalone Application nature since it would make it less readable IMO

### CODE REVIEW

File: ApplicationDeveloperGuide/moduleNatures.rst [hunk 1]
```diff
@@ -124,6 +124,51 @@ This module nature defines the following dedicated configuration properties:
         Possible values are `emb` (only on Device), `sim` (only Simulator) and `common` (both).
       - ``common``
 
+.. _module_natures.kernel:
+
+Kernel
+------
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/moduleNatures.rst:130]:
```suggestion
Kernel Application
------------------
```

File: ApplicationDeveloperGuide/moduleNatures.rst [hunk 2]
```diff
@@ -124,6 +124,51 @@ This module nature defines the following dedicated configuration properties:
...
+   :widths: 25 65 15
+   :header-rows: 1
+
+   * - Name
+     - Description
+     - Default
+   * - application.main.class
+     - Full Qualified Name of the main class of the kernel. This option is required.
+     - Not set
+   * - runtime.api.name
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/moduleNatures.rst:159]: can you double check with @dalgassimou 
I though these options were mandatory unless you provide a runtime API module

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/moduleNatures.rst (hunk 1 above)
Comment (tdelhomenie-microej) [reply ApplicationDeveloperGuide/moduleNatures.rst]: Done

File: ApplicationDeveloperGuide/moduleNatures.rst (hunk 2 above)
Comment (dalgassimou) [reply ApplicationDeveloperGuide/moduleNatures.rst:159]: The build-type firmware-multiapp is capable of loading a properties file found in the `kernelapi` conf.
And our runtime api (developer) provide a properties file, so yes the properties are mandantory unless you have a dependency in `kernelapi` conf which provides a properties file. and of course if this file contains the right properties.

**Review by frivieremicroej — APPROVED** [review]

**Pull request merged**